lib/protobuf/enum.rb

```diff
@@ -120,7 +127,8 @@ def self.enum_for_name(name)
     #   Enums, the first enum defined will be returned.
     #
     def self.enum_for_tag(tag)
-      enums_for_tag(tag).first
+      value = mapped_enums[tag.to_i]
+      value ? value.first : nil
```

Comment:
No because the spec allows aliasing enums:

> You can define aliases by assigning the same value to different enum constants. To do this you need to set the allow_alias option to true, otherwise protocol compiler will generate an error message when aliases are found.

source: https://developers.google.com/protocol-buffers/docs/proto?hl=en#enum